libbeat/publisher/pipeline/acker.go

```diff
@@ -133,25 +133,31 @@ func (a *gapCountACK) ackLoop() {
 
 	acks, drop := a.acks, a.drop
 	closing := false
+	empty := false
```

Comment:
the empty variable is only used in the `case n := <-acks` block. As the state is changed asynchronously the 'empty' variable is MUST only be used after a call the handleACK and if `closing` is true. Let's reduce the scope where `empty` is available to the block where it is actually initialised and used.